packages/iov-bns/src/bnsconnection.ts

```diff
@@ -566,7 +573,7 @@ export class BnsConnection implements AtomicSwapConnection {
    * It returns a stream starting the array of all existing transactions
    * and then continuing with live feeds
    */
-  public liveTx(query: TransactionQuery): Stream<ConfirmedTransaction | FailedTransaction> {
+  public liveTx(query: TransactionQuery): Stream<ConfirmedTransaction<LightTransaction> | FailedTransaction> {
```

Comment:
Now you can deduplictae the implementation by using `this.liveTxUnsigned`; same for `listenTx` and `searchTx`

Reply:
We still have this problem with streams though, don't we?

Reply:
We cannot use the result 1:1, but a simple map simipar to

```ts
this.liveTxUnsigned.map((original): ConfirmedTransaction<LightTransaction> | FailedTransaction) => {
   return original;
});```

should do the job